### PR TITLE
Feature screenshot all views

### LINKED ISSUE #541: Ability of take a screenshot of all the devices at once
# 🚀 Feature Request

### 📝 Description
I tend to share screenshots of all the devices at once with my team. Currently I just use a screenshot tool but it would be nice to be able to have it done by the app. As example
image

### ✨ Describe the solution you'd like

The app could create a screenshot of it's content and scale it.

### ✍️ Describe alternatives you've considered

Just using the screenshots app from my OS

### LINKED ISSUE #917: Screenshot all preview devices

### PR DESCRIPTION
**Pull request by durgakiran**
# ✨ Pull Request
Allows screen shot of all the screens

### 📓 Referenced Issue
Fixes #917 

### ℹ️ About the PR
- Adds Device name to the screenshot 
- Gives a button to capture screenshot of all devices
- Users existing screenshot flow to capture screens

### CODE REVIEW

File: desktop-app/src/renderer/components/ToolBar/index.tsx [hunk 1]
```diff
@@ -44,10 +99,22 @@ const ToolBar = () => {
...
+        <Icon icon="lucide:camera" />
+      </Button>
       <ColorSchemeToggle />
       <Divider />
       <PreviewSuiteSelector />
       <Menu />
+      <ModalLoader
+        isOpen={isCapturingScreenshot}
+        onClose={handleClose}
+        title="loading..."
```
Comment (manojVivek) [inline desktop-app/src/renderer/components/ToolBar/index.tsx:116]:
```suggestion
        title="Screenshot"
```

**Review by manojVivek — CHANGES_REQUESTED** [review]
@durgakiran Thanks for implementing this feature. 🙌🏼 

Added a few minor code review comments. Also, we need to change the PR description a bit as this PR doesn't close #541, but a clear prerequisite for that one. 

When you're fixing the comment, please sign the CLA as well. 

I think, we can merge this with the current scope and then add the canvas output as the next iteration. Wdyt?

File: desktop-app/src/renderer/components/ToolBar/index.tsx [hunk 2]
```diff
@@ -1,25 +1,80 @@
...
+  const screenshotCaptureHandler = async () => {
+    dispatch(setIsCapturingScreenshot(true));
+    const webViews: NodeListOf<Electron.WebviewTag> =
+      document.querySelectorAll('webView');
+    const screens: Array<ScreenshotAllArgs> = [];
+    const devices = activeSuit.devices.map((d) => getDevicesMap()[d]);
+    webViews.forEach(async (webview) => {
+      const device = devices.find((d) => d.name === webview.id);
+      const webPage = new WebPage(webview as unknown as Electron.WebContents);
+      const pageHeight = await webPage.getPageHeight();
```
[the remark this answers is not in the document]
Comment (durgakiran) [reply desktop-app/src/renderer/components/ToolBar/index.tsx:41]: This is not common, as this loop is specific to screenshot all functionality

File: desktop-app/src/renderer/components/ToolBar/index.tsx (hunk 1 above)
Comment (durgakiran) [reply desktop-app/src/renderer/components/ToolBar/index.tsx]: Updated

File: desktop-app/src/renderer/components/ToolBar/index.tsx [hunk 3]
```diff
@@ -1,25 +1,80 @@
...
+  const isCapturingScreenshot = useSelector(selectIsCapturingScreenshot);
+  const activeSuit = useSelector(selectActiveSuite);
   const dispatch = useDispatch();
 
+  const screenshotCaptureHandler = async () => {
+    dispatch(setIsCapturingScreenshot(true));
+    const webViews: NodeListOf<Electron.WebviewTag> =
+      document.querySelectorAll('webView');
+    const screens: Array<ScreenshotAllArgs> = [];
+    const devices = activeSuit.devices.map((d) => getDevicesMap()[d]);
```
[the remark this answers is not in the document]
Comment (durgakiran) [reply desktop-app/src/renderer/components/ToolBar/index.tsx]: Thanks for identifying type. Updated.

File: desktop-app/src/renderer/components/ToolBar/index.tsx [hunk 4]
```diff
@@ -1,25 +1,80 @@
...
+import ModalLoader from '../ModalLoader';
 import { PreviewSuiteSelector } from './PreviewSuiteSelector';
 
 const Divider = () => <div className="h-6 w-px bg-gray-300 dark:bg-gray-700" />;
 
 const ToolBar = () => {
   const rotateDevice = useSelector(selectRotate);
   const isInspecting = useSelector(selectIsInspecting);
+  const isCapturingScreenshot = useSelector(selectIsCapturingScreenshot);
+  const activeSuit = useSelector(selectActiveSuite);
```
[the remark this answers is not in the document]
Comment (durgakiran) [reply desktop-app/src/renderer/components/ToolBar/index.tsx]: Thanks for identifying type, updated.

**Comment by durgakiran** [comment]
> @durgakiran Thanks for implementing this feature. 🙌🏼
> 
> Added a few minor code review comments. Also, we need to change the PR description a bit as this PR doesn't close #541, but a clear prerequisite for that one.
> 
> When you're fixing the comment, please sign the CLA as well.
> 
> I think, we can merge this with the current scope and then add the canvas output as the next iteration. Wdyt?

Yes, That was my intention. Also, I have been working on the merge functionality with canvas output. Will soon raise but don't want to clutter to many things in one PR.

File: desktop-app/src/renderer/components/ToolBar/index.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (manojVivek) [reply desktop-app/src/renderer/components/ToolBar/index.tsx:41]: No, no, the loop is fine. 
I'm talking about the height and zoom adjusting logic from the single screenshot here: https://github.com/responsively-org/responsively-app/blob/main/desktop-app/src/renderer/components/Previewer/Device/Toolbar.tsx#L84-L90

[the remark this answers is not in the document]
Comment (durgakiran) [reply desktop-app/src/renderer/components/ToolBar/index.tsx:41]: oh ok

[the remark this answers is not in the document]
Comment (durgakiran) [reply desktop-app/src/renderer/components/ToolBar/index.tsx:42]: updated

**Review by manojVivek — APPROVED** [review]
Looking good! And thanks for your patience with the review!

**Pull request merged**